tools/azure-sdk-tools/ci_tools/build.py

```diff
@@ -189,3 +189,17 @@ def create_package(
 
     if enable_sdist:
         run([sys.executable, "setup.py", "sdist", "-d", dist], cwd=setup_parsed.folder, check=True)
+
+        # The file normalization changes in setuptools 67.3.0 intends source distribution file names to have the
+        # formatting a whl distributions. This patch avoids that for now.
+        normalized_package_name = setup_parsed.name.replace("-", "_")
+        package_for_adjustments = [f for f in os.listdir(dist) if normalized_package_name in f and f.endswith("tar.gz")]
```

Comment:
Swapping this to a more specific find. If we just use the name `azure-core-experimental` will also get messed up.